### DISCUSSION TITLE
ESP32-S2 New Release: Awesome! and SPIFFS question

### DISCUSSION THREAD

**Discussion by Dresch123** [Bodmer/TFT_eSPI#1528] [New processor/display support or capability request]
Hi Bodmer,
The new release for the ESP32-S2 works great with the WEMOS S2 Mini. Thank you very much!
One question:
Do you know of any way to upload data files to SPIFFS on ESP32-S2 based boards? "ESP32 Sketch Data Upload" in the Arduino IDE errors out when it finds the ESP32-S2 is being used.
Any thoughts or suggestions would be appreciated.
Thanks, Bill
Another use for your SPIFFS+JPEG example:
https://www.instructables.com/Enclosure-for-Your-TTGO-T-Display-Project/

![Round2](https://user-images.githubusercontent.com/3015603/147958165-603a1581-ccea-41e1-a080-86c101fb91f8.jpg)
![Round1](https://user-images.githubusercontent.com/3015603/147958059-1117d29c-8390-4ab2-a0b0-9d1444846c03.jpg)

**Comment by Bodmer** [comment]
I suspect SPIFFS is not supported for the ESP32 S2 or there is an update to the IDE SPIFFS loader.

I thnk that LittleFS is now preffered due to the higher perfomance.

**Comment by Bodmer** [comment]
I like the retro TV project!

**Comment by Dresch123** [comment]
Thanks, I will look into LittleFS to see if there is a path to uploading files to serial FLASH memory.
One other question: Is there a way of using your JPEG library to decode a JPEG file to a full size data array and then using pushImage to display the decoded file? Using the JPEG decoder to image directly to screen produces some odd motion artifacts when running video frames (not sure if the attached recording highlights it). My thought is that if I could buffer a whole image and then display it, the artifacts would decrease... at of course a hit to overall frame rate. 
Thanks again for the wonderful libraries and examples.
Bill
https://user-images.githubusercontent.com/3015603/148431635-2f36c748-3e12-4d5f-ae3c-1f49a619888b.mp4

![240x240](https://user-images.githubusercontent.com/3015603/148431641-21c4da15-717e-4075-8e3a-270d0400ce62.jpg)

**Comment by Dresch123** [comment]
https://forum.arduino.cc/t/uploading-littlefs-data-to-flash/664100

Ah... OK, will try it.

**Comment by Bodmer** [comment]
You can render the jpeg in a sprite then push the sprtie to the screen, a little example is here - ignore the rotation part:

https://github.com/Bodmer/TFT_eSPI/blob/master/examples/Sprite/Rotated_Sprite_3/Rotated_Sprite_3.ino

**Reply by Dresch123** [reply]
Hi,
I don't know if you got pinged on the reply below. It appears TFT_eFEX has some problems compiling with ESP32S2 as the selected processor. Thanks,
Bill

**Comment by Dresch123** [comment]
Thank you! I will give it a try. Bill

**Reply by Dresch123** [reply]
![NativeDecoderS2](https://user-images.githubusercontent.com/3015603/148699770-3e5ebc9b-c13a-41b3-a7fb-5e7960318355.jpg)
Hi,
In attempting to use the Sprite technique I ran into a compile error (see above) which I think is related to using the ESP32S2 and the native ESP32 JPEG decoder. This file: rom/tjpgd.h can't be found. This seems to be happening in the TFT_eFEX library.
Any suggestions? Thanks, Bill

**Reply by Dresch123** [reply]
The file does compile without issue for the normal ESP32 Dev Module selection so the problem most likely lies with the ESP32S2 selection and the TFT_eFEX library.

One other thought. Since a sprite is limited to about 80K, how difficult would it be to create a JPEG to TWO sprite routine? For instance, create UpperSprite and Lower Sprite and each would be the size of half a display (320 x 120 or 240 x 120, for examples). The JPEG would get decoded evenly between the two Sprites. When the routine returns, you could do two pushImages to completely fill the screen. Sort of like you do with that Bounce INO with the floating numbered circles. I know this is kind of a specialized routine but since a great number of screens are 320 x 240 or 240 x 240, this would handle a lot interesting use cases.

Thanks again Bodmer. The libraries you have created are amazing!

**Reply by Bodmer** [reply]
If there is a problem with TFT_eFEX then please raise an issue on that repository.

**Reply by Dresch123** [reply]
Will do. Thank you again. Bill

**Comment by Dresch123** [comment]
A better thought: A JPEG decoder routine that decodes a section of the JPEG based on a call like this:
bool JPEG_Section_Decoder(inputJPEG, numberofsections, whichsection, outputarray);
For example:
JPEG_Section_Decoder(panda, 4, 0, panda_out);
This would put the first 25% of the panda jpeg into the array panda_out.

Not sure any of this is possible with the JPEG structure, but if it was, this would be useful.

**Reply by Bodmer** [reply]
This is already possible. A jpeg must always be decoded from the start, but decoding can be stopped at any time.  The start y coord for rendering a jpeg can be negative, so the bottom part of a jpeg can be captured on screen or in a sprite.

**Reply by Dresch123** [reply]
Thanks! I will have to try this.
